pkg/doc/did/doc.go

```diff
@@ -79,7 +79,7 @@ func (d *DID) String() string {
 func Parse(did string) (*DID, error) {
 	// I could not find a good ABNF parser :(
 	const idchar = `a-zA-Z0-9-_\.`
-	regex := fmt.Sprintf(`^did:[a-z0-9]+:(:+|[:%s]+)*[%s]+$`, idchar, idchar)
+	regex := fmt.Sprintf(`^did:[a-z0-9]+:(:+|[:%s]+)*[%%:%s]+[^:]$`, idchar, idchar)
```

Comment:
what did you fix here?

Reply:
was it due to the url-encoding?

Reply:
yes that was precisely it, however as you mentioned in https://github.com/w3c-ccg/did-method-web/issues/7#issuecomment-730434663 this isn't exactly valid for the time being, looking into workarounds as we speak.

Reply:
Followup: #2336 

We should fix `did.Parse()`